src/cml.js

```diff
@@ -0,0 +1,103 @@
+const { execSync } = require('child_process');
+const git_url_parse = require('git-url-parse');
+
+const Gitlab = require('./drivers/gitlab');
+const Github = require('./drivers/github');
+const { upload, exec } = require('./utils');
+
+const uri_no_trailing_slash = (uri) => {
+  return uri.endsWith('/') ? uri.substr(0, uri.length - 1) : uri;
+};
+
+const repo_from_origin = () => {
+  const origin = execSync('git config --get remote.origin.url').toString(
+    'utf8'
+  );
+  return git_url_parse(origin).toString('https').replace('.git', '');
+};
+
+const infer_driver = (opts = {}) => {
+  const { repo } = opts;
+  if (repo && repo.includes('github.com')) return 'github';
+  if (repo && repo.includes('gitlab.com')) return 'gitlab';
+
+  const { GITHUB_REPOSITORY, CI_PROJECT_URL } = process.env;
+  if (GITHUB_REPOSITORY) return 'github';
+  if (CI_PROJECT_URL) return 'gitlab';
```

Comment:
Do we need similar variables for GL? Why not?

Reply:
I'm a bit confused by `CI_PROJECT_URL` :) Why it is not just GitLab like `GITLAB_REPOSITORY `?

Reply:
CI_PROJECT_URL is an env variable that GL CI generates. Its not our choose to name it that way. What we are doing here is try to guess if the driver is not set nor the repo either where are we being run. 
This is useful for on premise instances also.

Reply:
What we should do if none is defined? Should we throw a meaningful exception?

Reply:
nothing! later on the exception will be raised when creating the Driver